core/src/main/scala/org/http4s/headers/Forwarded.scala

```diff
@@ -0,0 +1,185 @@
+
+      def apply(address: Uri.Ipv4Address): Name = Name.Ipv4(address)
+      def apply(address: Inet4Address): Name = apply(Uri.Ipv4Address.fromInet4Address(address))
+      def apply(a: Byte, b: Byte, c: Byte, d: Byte): Name = apply(Uri.Ipv4Address(a, b, c, d))
+
+      def apply(address: Uri.Ipv6Address): Name = Name.Ipv6(address)
+      def apply(address: Inet6Address): Name = apply(Uri.Ipv6Address.fromInet6Address(address))
+      def apply(a: Short, b: Short, c: Short, d: Short, e: Short, f: Short, g: Short, h: Short)
+          : Name = apply(Uri.Ipv6Address(a, b, c, d, e, f, g, h))
+    }
+
+    sealed trait Port { _: Product => }
+
+    object Port {
+      private[this] final case class C(value: Int) extends Port {
+        override def productPrefix: String = "Port"
+      }
+
+      def fromInt(num: Int): ParseResult[Port] =
+        checkPortNum(num).toLeft(C(num))
+
+      def unapply(port: Port): Option[Int] =
+        PartialFunction.condOpt(port) {
+          case C(num) => num
+        }
+    }
+
+    sealed trait Obfuscated extends Name with Port { _: Product =>
+
+      /**
+        * Obfuscated value must start with '_' (underscore) symbol.
+        */
+      def value: String
+    }
+    object Obfuscated {
+      private[this] final case class C(value: String) extends Obfuscated {
+        override def productPrefix: String = "Obfuscated"
+      }
+
+      def fromString(s: String): ParseResult[Obfuscated] =
+        new ModelNodeObfuscatedParser(s).parse
+
+      def unapply(o: Obfuscated): Option[String] = Some(o.value)
+
+      // Referenced by model parsers.
+      private[http4s] def apply(s: String): Obfuscated = C(s)
+    }
+
+    def fromString(s: String): ParseResult[Node] = new ModelNodeParser(s).parse
+  }
+
+  sealed trait Host { _: Product =>
+    def host: Uri.Host
+    def port: Option[Int]
+  }
+
+  object Host {
+    private[this] final case class C(host: Uri.Host, port: Option[Int]) extends Host {
+      override def productPrefix: String = "Host"
+    }
+
+    def apply(uriHost: Uri.Host): Host = C(uriHost, None)
+
+    def from(uriHost: Uri.Host, port: Int): ParseResult[Host] =
+      checkPortNum(port).toLeft(C(uriHost, Some(port)))
+
+    def from(uriHost: Uri.Host, port: Option[Int]): ParseResult[Host] =
+      port.fold(apply(uriHost).asRight[ParseFailure])(from(uriHost, _))
+
+    def fromString(s: String): ParseResult[Host] = new ModelHostParser(s).parse
+
+    def fromUri(uri: Uri): ParseResult[Host] =
+      uri.host
+        .map(C(_, uri.port))
+        .toRight(ParseFailure("missing host", s"no host defined in the URI '$uri'"))
+
+    def unapply(host: Host): Option[(Uri.Host, Option[Int])] = Some((host.host, host.port))
+  }
+
+  type Proto = Uri.Scheme
+  val Proto: Uri.Scheme.type = Uri.Scheme
+
+  sealed trait Element extends Product {
+    def `by`: Option[Node]
+    def `for`: Option[Node]
+    def `host`: Option[Host]
+    def `proto`: Option[Proto]
```

Comment:
Back-ticks are here only because of `for` – apparently we cannot use it as an identifier.
All other field names don't need back-ticks (apparently), but I found it would be ugly to back-tick `for` only. But it looks quite ugly anyway. So I'd appreciate any better naming suggestions here:)